server/api/crud/alerts.py

```diff
@@ -0,0 +1,269 @@
+
+                if state_obj is None:
+                    state_obj = {"events": [event_data.timestamp]}
+
+                if alert.criteria.period is not None:
+                    state_obj["events"].append(event_data.timestamp)
+                    # adjust the sliding window of events
+                    self._normalize_events(
+                        state_obj, self._string2datetime(alert.criteria.period)
+                    )
+
+                if len(state_obj["events"]) >= alert.criteria.count:
+                    send_notification = True
+            else:
+                send_notification = True
+
+            active = False
+            if send_notification:
+                state.count += 1
+                logger.debug(f"Sending notifications for alert {alert.name}")
+                AlertNotificationPusher().push(alert, event_data)
+
+                if alert.reset_policy == "auto":
+                    self.reset_alert(session, alert.project, alert.name)
+                else:
+                    active = True
+
+            server.api.utils.singletons.db.get_db().store_alert_state(
+                session,
+                alert.project,
+                alert.name,
+                count=state.count,
+                last_updated=event_data.timestamp,
+                obj=state_obj,
+                active=active,
+            )
+
+    @staticmethod
+    def _validate_alert(alert, name, project):
+        if name != alert.name:
+            raise mlrun.errors.MLRunBadRequestError(
+                f"Alert name mismatch for alert {name} for project {project}. Provided {alert.name}"
+            )
+
+        if (
+            alert.criteria is not None
+            and alert.criteria.period is not None
+            and Alerts._string2datetime(alert.criteria.period) is None
+        ):
+            raise mlrun.errors.MLRunBadRequestError(
+                f"Invalid period ({alert.criteria.period}) specified for alert {name} for project {project}"
+            )
+
+        for notification in alert.notifications:
+            if notification.kind not in [
+                mlrun.common.schemas.NotificationKind.git,
+                mlrun.common.schemas.NotificationKind.slack,
+                mlrun.common.schemas.NotificationKind.webhook,
+            ]:
+                raise mlrun.errors.MLRunBadRequestError(
+                    f"Unsupported notification ({notification.kind}) for alert {name} for project {project}"
+                )
+            notification_object = mlrun.model.Notification.from_dict(
+                notification.dict()
+            )
+            notification_object.validate_notification()
+
+        if alert.entity.project != project:
+            raise mlrun.errors.MLRunBadRequestError(
+                f"Invalid alert entity project ({alert.entity.project}) for alert {name} for project {project}"
+            )
+
+    @staticmethod
+    def _string2datetime(date_str):
```

Comment:
More accurate name
```suggestion
    def _string_to_timedelta(date_str):
```
And maybe a good idea to move this to a `mlrun.common.utils.helpers`